c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -4444,17 +4615,69 @@ bool process_next_body_read(JSContext *cx) {
   return true;
 }
 
-bool process_network_io(JSContext *cx) {
-  if (!has_pending_requests())
-    return true;
+bool process_pending_async_tasks(JSContext *cx) {
+  MOZ_ASSERT(has_pending_async_tasks());
+
+  uint32_t timeout = 0;
+  if (!timers->empty()) {
```

Comment:
Is it possible for a timer that repeatedly sets another timer with delay `0` to starve the loop that processes `pending_async_tasks`?

Reply:
I don't believe it would starve the event loop, I ran the below in Firefox's console and could run other functions after it:
```js
function starve() {
console.log(2)
setTimeout(starve, 0)
}
starve()
console.log(1)
```

I'll see if I can come up with a test C@E application for this

Reply:
This explains why setTimeout with a duration of 0 doesn't starve the loop -- https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop#zero_delays

Reply:
You can starve the loop with Promises though :D
```js
function starveWithPromises() {
console.log(3)
Promise.resolve().then(starveWithPromises)
}
starveWithPromises()
console.log(9) // this will never run
```